### ISSUE TITLE
etwas "Code-Kritik", insbesondere bzgl. Verwendung von `Task.perform ... ... (succeed ...)`

### ISSUE THREAD

**Issue by jvoigtlaender** [Sulring/elmaction#8]
Wann immer Sie etwas wie `Task.perform ... function (succeed value)` machen, wäre m.E. besser: `Cmd.Extra.message (function value)`, mit `message` aus http://package.elm-lang.org/packages/shmookey/cmd-extra. Schon weil Sie dann nicht eine willkürliche, und an den meisten Stellen sinnfreie, Funktion für das `...` einsetzen müssen. (Sie verwenden an den meisten Stellen `SoundError` dafür, obwohl die jeweiligen Messages gar nichts mit Sound zu tun haben.)

In der Regel fände ich sogar einen Verzicht auf den "Umweg" über ein `Cmd` sinnvoller, stattdessen direkten rekursiven Aufruf von `update`. Also, mal eine relativ zufällige Stelle:
```elm
update : Msg -> Model -> ( Model, Cmd Msg )
update action model =
    case action of
        ...

        FetchFail err ->
            case err of
                Http.Timeout -> (model, Task.perform SoundError ChangeStatus (succeed (log "I failed of timeout" Highscore)))
```
würde ich stattdessen schreiben als:
```elm
        FetchFail err ->
            case err of
                Http.Timeout -> update (ChangeStatus (log "I failed of timeout" Highscore)) model
```
Selbst an Stellen wo Sie `Cmd.batch` verwenden und mehrere solche Messages-as-Commands verknüpfen und zurückgeben, wäre rekursives Update vermutlich besser und sogar klarer, insbesondere wenn Sie Helfer aus http://package.elm-lang.org/packages/ccapndave/elm-update-extra verwenden.

Und schließlich, nicht unabhängig von Obigem, ist mir nicht klar, ob Ihre Modellierung mittels `Cmd` in den Typen `OnActorKilled` und `OnAction` sowie im Feld `fireCommand` von `Actor` sinnvoll ist. Könnte da nicht eventuell überall `Cmd Msg` durch `List Msg` ersetzt werden? (Weil Sie zum Bauen der entsprechenden `Cmd`s sowieso nur `Cmd.none`, `Cmd.batch` und - im Prinzip - `Cmd.Extra.message` verwenden, also nur eine Liste von `Msg`s aufbauen, was Sie genauso gut auch direkt machen könnten, um sie dann per `List.foldl` mit `update` abzuspulen.)

**Comment by Sulring** [comment]
Stimmt! Anscheinend habe ich den offensichtlichsten Weg übersehen. Ich war nie mit  `Task.perform SoundError ...`  zufrieden, dachte aber gar nicht um einer alternativen Lösung nach.
